test/helpers/kubectl.go

```diff
@@ -4077,3 +4077,21 @@ func logGathererSelector(allNodes bool) string {
 
 	return selector
 }
+
+// GetDNSProxyPort returns the port the Cilium DNS proxy is listening on
+func (kub *Kubectl) GetDNSProxyPort(ciliumPod string) int {
+	const pickDNSProxyPort = `iptables-save -t mangle | sed -n '/udp.*TPROXY to host cilium-dns-egress/ s/.*--on-port \([1-9][0-9]*\).*/\1/p'`
+
+	// Find out the DNS proxy ports in use
+	res := kub.CiliumExecContext(context.TODO(), ciliumPod, pickDNSProxyPort)
+	if !res.WasSuccessful() {
+		ginkgoext.Failf("Cannot find DNS proxy port on %s", ciliumPod)
+	}
+	portStr := res.GetStdOut().String()
+	gomega.ExpectWithOffset(1, portStr).ShouldNot(gomega.BeEmpty(), "No DNS proxy port found on %s", ciliumPod)
+	port, err := strconv.Atoi(strings.TrimSpace(portStr))
+	if err != nil || port == 0 {
+		ginkgoext.Failf("Invalid DNS proxy port on %s: %s", ciliumPod, portStr)
+	}
+	return port
```

Comment:
Can we pull this from `cilium status` rather than the iptables rules? In future we won't rely on iptables so it'd be nice to have tests that don't rely on this implementation detail.

Reply:
`cilium status` does not currently report the actual proxy ports, so this would be a clean-up for a later PR.

Reply:
Not true :-)

```
$ kubectl -n kube-system exec -ti pod/cilium-6nbc5 -- cilium status -o json | jq -r '.proxy.redirects[0]."proxy-port"'
11128
```

Reply:
(I'm fine to have this as an independent follow-up though)

Reply:
OK, I tried it without any policy imported (but DNS proxy still running) and in that case there are no redirects (`.proxy.redirects == null`), but the bug still exists as the DNS proxy is already listening.